### PR TITLE
chore: refactor scan mode model, controller, services and tests

### PR DESCRIPTION
**Pull request by burgerni10**

### CODE REVIEW

File: backend/src/service/scan-mode.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    return this.scanModeRepository.findAll();
+  }
+
+  findById(id: string): ScanMode | null {
+    const scanMode = this.scanModeRepository.findById(id);
+    if (scanMode) {
+      return scanMode;
+    } else {
+      return null;
+    }
```
Comment (marouanehassanioptimistik) [inline backend/src/service/scan-mode.service.ts:30]: this is weird this should already be the return type of the repo method, you can just proxy to the repo call

File: backend/src/service/scan-mode.service.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  async delete(scanModeId: string): Promise<void> {
+    const scanMode = this.scanModeRepository.findById(scanModeId);
+    if (!scanMode) {
+      throw new Error(`Scan mode ${scanModeId} not found`);
+    }
+
+    this.scanModeRepository.delete(scanModeId);
+    this.southCacheRepository.deleteAllByScanMode(scanModeId);
```
Comment (marouanehassanioptimistik) [inline backend/src/service/scan-mode.service.ts:61]: do you have a foreign key? if yes then maybe the order matters not sure but this require a closer look

**Review by marouanehassanioptimistik — CHANGES_REQUESTED** [review]
overall this is way better let's continue on this track some entities are more complex

File: backend/src/model/scan-mode.model.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export const toScanModeDTO = (scanMode: ScanMode): ScanModeDTO => {
+  return {
+    id: scanMode.id,
+    name: scanMode.name,
+    description: scanMode.description,
+    cron: scanMode.cron
+  };
+};
```
[the remark this answers is not in the document]
Comment (burgerni10) [reply backend/src/model/scan-mode.model.ts:23]: TODO: place it in the service

**Pull request merged**